pkg/pool-stable-phantom/contracts/StablePoolAmplification.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/WordCodec.sol";
+
+import "./StableMath.sol";
+
+abstract contract StablePoolAmplification is Authentication {
```

Comment:
Make this inherit from `BasePoolAuthorization` instead of plain Authentication, and override `isOwnerOnly` for the two amp functions.

Reply:
It seems to need BasePool (for the inheritance graph to work out)

Reply:
Per discussion, the problem was super; removing the call to super lets me derive from BasePoolAuthorization